pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -582,9 +586,18 @@ func (cb *ClusterBuilder) setUseDownstreamProtocol(mc *clusterWrapper) {
 		mc.httpProtocolOptions = &http.HttpProtocolOptions{}
 	}
 	options := mc.httpProtocolOptions
+	// Inherit the HTTP1.x protocol options from the explicit HTTP1.x config.
+	explicitHTTP1Options := &core.Http1ProtocolOptions{}
+	if explicitConfig, ok := options.GetUpstreamProtocolOptions().(*http.HttpProtocolOptions_ExplicitHttpConfig_); ok {
```

Comment:
What happens if this type assertion fails? Do we log anywhere? Also, in what circumstances does `GetUpstreamProtocolOptions` return explicit config vs. common config?

Reply:
If it fails, the `HttpProtocolOptions` in the `HttpProtocolOptions_UseDownstreamHttpConfig` will be set to the empty config `&core.Http1ProtocolOptions{}` from line 590. Before this change was introduced `HttpProtocolOptions` was also set to the empty config `&core.Http1ProtocolOptions{}`

My thought process here is when `port.Protocol.IsUnsupported()` is set along with `preserveHeaderCase` the use downstream protocol settings should inherit existing http1 protocol options if they exist instead of being overwritten with an empty config `&core.Http1ProtocolOptions{}`

Reply:
Ack, makes sense